api/src/main/java/org/openmrs/module/radiology/impl/RadiologyServiceImpl.java

```
@@ -127,18 +141,86 @@ public RadiologyOrder placeRadiologyOrder(RadiologyOrder radiologyOrder) {
 	 * @param provider the encounter provider
 	 * @param encounterDateTime the encounter date
 	 * @return radiology order encounter for given parameters
-	 * @should save radiology order encounter for given parameters
+	 * @should create encounter for radiology order for given parameters
+	 * @should create encounter for new radiology order for patient with existing visit
+	 * @should create encounter for new radiology order for patient without existing visit
+	 * @should throw illegal state exception if encounter cannot be created
 	 */
 	@Transactional
-	private Encounter saveRadiologyOrderEncounter(Patient patient, Provider provider, Date encounterDateTime) {
+	private Encounter saveRadiologyOrderEncounter(Patient patient, Provider provider, Date encounterDateTime)
+	        throws IllegalStateException {
+		try {
+			EncounterTransaction encounterTransaction = setUpEncounterTransaction(patient, encounterDateTime,
+			    radiologyProperties.getRadiologyVisitType(), radiologyProperties.getRadiologyOrderEncounterType(), provider,
+			    radiologyProperties.getOrderingProviderEncounterRole());
+			encounterTransaction = emrEncounterService.save(encounterTransaction);
+			return encounterService.saveEncounter(encounterService.getEncounterByUuid(encounterTransaction
+			        .getEncounterUuid()));
+		}
+		catch (Exception e) {
```

@bgeVam dont catch all exceptions, only the ones you care about